addons/base_automation/models/base_automation.py

```diff
@@ -498,7 +498,13 @@ def patch(model, name, method):
             """ Patch method `name` on `model`, unless it has been patched already. """
             if model not in patched_models[name]:
                 patched_models[name].add(model)
-                model._patch_method(name, method)
+                ModelClass = type(model)
+                origin = getattr(ModelClass, name)
+                method.origin = origin
+                # propagate decorators from origin to method, and apply api decorator
+                wrapped = api.propagate(origin, method)
+                wrapped.origin = origin
+                setattr(ModelClass, name, wrapped)
```

Comment:
All the methods already have the right decorator in this case.
So IMO there is no need for propagation and we can simplify the code:
```suggestion
                ModelClass = type(model)
                method.origin = getattr(ModelClass, name)
                setattr(ModelClass, name, method)
```

Reply:
> All the methods already have their right decorator in this case.

What about the `create`? The `api.returns` decorator is not present in the `make_create` then the `create._returns` will disappear after the patch. Firstly, I think that the return value in the RPC was no longer correct.
But we have a special case for the create method: https://github.com/odoo/odoo/blob/3353dfdb297cd020525b0ec64c81f5af5ff4ca50/odoo/api.py#L440
It seems that the `api.returns` is useless for the `create` method of models.py. I will remove it to avoid confusion in the future. Are you OK with that?

Reply:
> It seems that the `api.returns` is useless for the `create` method of models.py. I will remove it to avoid confusion in the future. Are you OK with that?

Indeed, it is useless :+1: